tokens/token-2022/transfer-hook/token-stop/anchor/programs/token-stop/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+use anchor_lang::prelude::*;
+
+use anchor_spl::{
+    token_interface::{
+        spl_token_metadata_interface::state::TokenMetadata, Mint,
+        Token2022, token_metadata_update_field, TokenMetadataUpdateField
+    },
+    token_2022::{
+        spl_token_2022::{
+            extension::{
+                BaseStateWithExtensions,
+                StateWithExtensions,
+            },
+            state::Mint as Token2022Mint
+        },
+        ID as TOKEN_2022_PROGRAM_ID,
+    },
+};
+
+use spl_tlv_account_resolution::{account::ExtraAccountMeta, state::ExtraAccountMetaList};
+use spl_transfer_hook_interface::instruction::ExecuteInstruction;
+
+declare_id!("Hy7xLUkd52wkE2xNYvNVYmmrLeqfkoqnw6sYK3cqbNqf");
+
+#[error_code]
+pub enum TransferStopError {
+    #[msg("The transfers have been disabled")]
+    TransfersDisabled,
+}
+
+#[program]
+pub mod token_stop {
+    use anchor_spl::token_interface::spl_token_metadata_interface::state::Field;
+
+    use super::*;
+
+    pub fn stop_transfer(ctx: Context<StopTransfer>) -> Result<()> {
+        let cpi_accounts = TokenMetadataUpdateField {
+            metadata: ctx.accounts.mint.to_account_info(),
+            update_authority: ctx.accounts.authority.to_account_info(),
+            token_program_id: ctx.accounts.token_program.to_account_info(),
+        };
+        let cpi_program = ctx.accounts.token_program.to_account_info();
+        let cpi_ctx = CpiContext::new(cpi_program, cpi_accounts);
+
+        token_metadata_update_field(
+            cpi_ctx,
+            Field::Key("disable-transfers".to_string()),
+            "true".to_string(),
+        )?;
+
+        Ok(())
+    }
+
+    pub fn resume_transfer(ctx: Context<ResumeTransfer>) -> Result<()> {
+        let cpi_accounts = TokenMetadataUpdateField {
+            metadata: ctx.accounts.mint.to_account_info(),
+            update_authority: ctx.accounts.authority.to_account_info(),
+            token_program_id: ctx.accounts.token_program.to_account_info(),
+        };
+        let cpi_program = ctx.accounts.token_program.to_account_info();
+        let cpi_ctx = CpiContext::new(cpi_program, cpi_accounts);
+
+        token_metadata_update_field(
+            cpi_ctx,
+            Field::Key("disable-transfers".to_string()),
+            "false".to_string(),
+        )?;
+
+        Ok(())
+    }
+
+    // we dont care about the token amount transferred
+    #[interface(spl_transfer_hook_interface::execute)]
+    pub fn execute(ctx: Context<Execute>/* , amount: u64 */) -> Result<()> {
+        let mint_info = ctx.accounts.mint.to_account_info();
+        let mint_data = mint_info.data.borrow();
+        let mint = StateWithExtensions::<Token2022Mint>::unpack(&mint_data)?;
+        
```

Comment:
You could add a check for good pratice to check if its actually currently transferring 
fn check_is_transferring(ctx: &Context<TransferHook>) -> Result<()> {
    let source_token_info = ctx.accounts.source_token.to_account_info();
    let mut account_data_ref: RefMut<&mut [u8]> = source_token_info.try_borrow_mut_data()?;
    let mut account = PodStateWithExtensionsMut::<PodAccount>::unpack(*account_data_ref)?;
    let account_extension = account.get_extension_mut::<TransferHookAccount>()?;

    if !bool::from(account_extension.transferring) {
        return err!(TransferError::IsNotCurrentlyTransferring);
    }

    Ok(())
}